test/e2e/suite/conformance/certificatesigningrequests/tests.go

```diff
@@ -166,25 +160,61 @@ func (s *Suite) Define() {
 			{
 				name:    "should issue a certificate that defines a Common Name and IP Address",
 				keyAlgo: x509.RSA,
-				csrModifiers: func() []gen.CSRModifier {
-					return []gen.CSRModifier{
-						gen.SetCSRCommonName("test-common-name-" + rand.String(10)),
-						gen.SetCSRIPAddresses(net.IPv4(127, 0, 0, 1), net.IPv4(8, 8, 8, 8)),
-					}
+				csrModifiers: []gen.CSRModifier{
+					gen.SetCSRCommonName("test-common-name-" + rand.String(10)),
+					gen.SetCSRIPAddresses(net.ParseIP(s.SharedIPAddress)),
 				},
 				kubeCSRUsages: []certificatesv1.KeyUsage{
 					certificatesv1.UsageDigitalSignature,
 					certificatesv1.UsageKeyEncipherment,
 				},
 				requiredFeatures: []featureset.Feature{featureset.CommonNameFeature, featureset.IPAddressFeature},
 			},
+			{
+				name:    "should issue a certificate that defines an IP Address",
+				keyAlgo: x509.RSA,
+				csrModifiers: []gen.CSRModifier{
+					gen.SetCSRIPAddresses(net.ParseIP(s.SharedIPAddress)),
+				},
+				kubeCSRUsages: []certificatesv1.KeyUsage{
+					certificatesv1.UsageDigitalSignature,
+					certificatesv1.UsageKeyEncipherment,
+				},
+				requiredFeatures: []featureset.Feature{featureset.IPAddressFeature},
+			},
+			{
+				name:    "should issue a certificate that defines a DNS Name and IP Address",
+				keyAlgo: x509.RSA,
+				csrModifiers: []gen.CSRModifier{
+					gen.SetCSRIPAddresses(net.ParseIP(s.SharedIPAddress)),
+					gen.SetCSRDNSNames(e2eutil.RandomSubdomain(s.DomainSuffix)),
+				},
+				kubeCSRUsages: []certificatesv1.KeyUsage{
+					certificatesv1.UsageDigitalSignature,
+					certificatesv1.UsageKeyEncipherment,
+				},
+				requiredFeatures: []featureset.Feature{featureset.OnlySAN, featureset.IPAddressFeature},
+			},
```

Comment:
Why do we need an E2E test for this?
Is important to know which issuers can reconcile a CertificateSiginingRequest containing only an IP address and DNS name?
And if so, should we hand craft an E2E test for every permutation? No, that would be absurd.

If anything we should just delete all these tests and replace them with a fuzz test which bombards cert-manager with 1000 permutations and see if there are any unexpected failures.

Reply:
This test is based on this test: https://github.com/cert-manager/cert-manager/blob/master/test/e2e/suite/conformance/certificates/tests.go#L461-L485

I propose we do a cleanup first and start thinking about fuzz testing once we have a clear view on these tests and exactly what specific test cases they verify.